gulpTasks/doc.js

```diff
@@ -24,40 +32,66 @@ gulp.task('doc', ['copyBinToDoc', 'buildPlayground'], function() {
   const src = app.expandInputFiles(['src']);
   const project = app.convert(src);
   app.generateDocs(project, 'docgen');
-  app.generateJson(project, './bin/docgen/docgen.json', 'https://coveo.github.io/search-ui/');
+  app.generateJson(project, docgenJsonPath, 'https://coveo.github.io/search-ui/');
   return gulp.src('./readme.png').pipe(gulp.dest('./docgen'));
 });
 
 gulp.task('cleanGeneratedThemesFiles', () => {
   return del(['./docs/theme/assets/gen/**/*']);
 });
 
-gulp.task('copyBinToDoc', ['cleanGeneratedThemesFiles'], function() {
+gulp.task('copyBinToDoc', ['cleanGeneratedThemesFiles'], () => {
   return gulp.src('./bin/{js,image,css}/**/*').pipe(gulp.dest('./docs/theme/assets/gen'));
 });
 
 gulp.task('buildPlayground', shell.task(['node node_modules/webpack/bin/webpack.js --config ./webpack.playground.config.js']));
 
-function copyFile(source, target, cb) {
-  var cbCalled = false;
+gulp.task('testDoc', () => {
+  const docgenJson = JSON.parse(fs.readFileSync(docgenJsonPath));
 
-  var rd = fs.createReadStream(source);
-  rd.on('error', function(err) {
-    done(err);
-  });
-  var wr = fs.createWriteStream(target);
-  wr.on('error', function(err) {
-    done(err);
-  });
-  wr.on('close', function(ex) {
-    done();
-  });
-  rd.pipe(wr);
+  // Check if object is defined and non-empty
+  if (!docgenJson || !docgenJson.length) {
+    throw new Error('Invalid object');
+  }
 
-  function done(err) {
-    if (!cbCalled) {
-      cb(err);
-      cbCalled = true;
+  // Check that all attributes are defined correctly
+  const attributes = [
+    { name: 'name', types: ['string'] },
+    { name: 'notSupportedIn', types: ['string', 'object'] },
+    { name: 'comment', types: ['string'] },
+    { name: 'type', types: ['string'] },
+    { name: 'constrainedValues', types: ['object'] },
+    { name: 'miscAttributes', types: ['object'] }
+  ];
+
+  _.each(docgenJson, element =>
+    _.each(attributes, ({ name, types }) => {
+      if (!_.has(element, name) || !_.contains(types, typeof element[name])) {
+        throw new Error(`Invalid or missing attribute "${name}" for doc element "${JSON.stringify(element)}"`);
+      }
+    })
+  );
+
+  // Check that some specific doc elements are present
+  const elements = [
+    {
+      name: 'SearchAlerts.usageAnalytics',
+      notSupportedIn: '',
+      comment:
```

Comment:
I'd change that to instead a "partial" contains match versus an exact match.

Some regex to check that it contains the keyword "Analytics" maybe.